src/test/java/com/checkmarx/flow/cucumber/component/webhook/WebHookSteps.java

```diff
@@ -0,0 +1,106 @@
+
+    public WebHookSteps(GitHubProperties properties) {
+        this.properties = properties;
+    }
+
+    @Given("CxFlow is running as a service")
+    public void runAsService() {
+        SpringApplication.run(CxFlowApplication.class, "--web");
+    }
+
+    @When("GitHub sends WebHook requests to CxFlow {int} times per second")
+    public void githubSendsWebHookRequests(int timesPerSecond) {
+        final int MAX_TOTAL_REQUESTS = 10,
+                MILLISECONDS_IN_SECOND = 1000;
+
+        int intervalBetweenRequestsMs = MILLISECONDS_IN_SECOND / timesPerSecond;
+
+        logger.info("Starting to send WebHook requests with the interval of {} ms.", intervalBetweenRequestsMs);
+        for (int i = 0; i < MAX_TOTAL_REQUESTS; i++) {
+            logger.info("Sending request #{}.", i + 1);
+            startRequestSendingTaskAsync();
+            chillOutFor(intervalBetweenRequestsMs);
+        }
+
+        waitForAllTasksToComplete(requestSendingTasks);
+    }
+
+    private static void chillOutFor(int durationMs) {
+        // Using Awaitility, because SonarLint considers Thread.sleep a code smell.
+        Awaitility.with()
+                .pollDelay(durationMs, TimeUnit.MILLISECONDS)
+                .await()
+                .until(() -> true);
+    }
+
+    private void startRequestSendingTaskAsync() {
+        CompletableFuture<Long> task = CompletableFuture.supplyAsync(this::getRequestExecutionTimeMs);
+        requestSendingTasks.add(task);
+    }
+
+    private long getRequestExecutionTimeMs() {
+        StopWatch stopWatch = new StopWatch();
+        stopWatch.start();
+        sendWebHookRequest();
+        return stopWatch.getTime();
+    }
+
+    private void sendWebHookRequest() {
+        throw new PendingException();
+    }
+
+    private void waitForAllTasksToComplete(List<CompletableFuture<Long>> tasks) {
+        logger.info("Waiting for all the requests to complete.");
+        CompletableFuture[] futureArray = tasks.toArray(new CompletableFuture[0]);
+        CompletableFuture<Void> combinedTask = CompletableFuture.allOf(futureArray);
+        Assert.assertFalse(combinedTask.isCompletedExceptionally());
+    }
+
+    @Then("each of the requests is answered in at most {int} ms")
+    public void eachOfTheRequestsIsAnsweredInAtMostMs(long expectedMaxDurationMs) {
+        Optional<Long> actualMaxDurationMs = requestSendingTasks.stream()
+                .map(WebHookSteps::toExecutionTimeMs)
+                .max(Long::compare);
+
+        Assert.assertTrue(actualMaxDurationMs.isPresent());
+
+        String message = String.format("Actual duration (%d ms) is greater than the expected max duration (%d ms).",
+                actualMaxDurationMs.get(),
+                expectedMaxDurationMs);
+
+        Assert.assertTrue(message, actualMaxDurationMs.get() <= expectedMaxDurationMs);
+    }
+
+    private static long toExecutionTimeMs(CompletableFuture<Long> task) {
+        try {
```

Comment:
what about the getexceptionaly()?

Reply:
Do you mean isCompletedExceptionally()? Even if I add it as a condition, I'll still need to use try..catch, so checking for isCompletedExceptionally() seems unnecessary here.